src/useVectorLayers.ts

```diff
@@ -6,52 +6,98 @@ export interface UseVectorLayersProps {
+    ...(truckRestrictions ? [{ feature: 'vehicle restrictions' }] : []),
+    ...(congestion
+      ? [{ feature: 'congestion zones' }, { feature: 'environmental zones' }]
+      : []),
+  ])
+}
+
 export const useVectorLayers = ({
   map,
   useSatellite,
   trafficLayer,
   congestion,
   truckRestrictions,
-  incidentsLayer,
   defaultLayers,
   useVectorTiles,
 }: UseVectorLayersProps) => {
   useEffect(() => {
-    if (map && defaultLayers && useVectorTiles) {
-      if (truckRestrictions) {
-        map.setBaseLayer(defaultLayers.vector.normal.truck)
-      } else {
-        map.setBaseLayer(useSatellite
-          ? defaultLayers.raster.satellite.map
-          : defaultLayers.vector.normal.map)
-      }
+    if (!map || !defaultLayers || !useVectorTiles) {
+      return
     }
-  }, [defaultLayers, truckRestrictions, congestion, map, useVectorTiles, useSatellite])
 
-  useEffect(() => {
-    if (map && defaultLayers && useVectorTiles) {
-      if (incidentsLayer) {
-        map.addLayer(defaultLayers.vector.normal.trafficincidents)
+    [
+      defaultLayers.vector.normal.logistics,
+      defaultLayers.hybrid.logistics.vector,
+    ].forEach(layer => {
+      const style = layer.getProvider().getStyleInternal() as H.map.render.harp.Style
+      if (style.getState() !== H.map.render.Style.State.READY) {
+        const changeListener = () => {
+          if (style.getState() === H.map.Style.State.READY) {
+            style.removeEventListener('change', changeListener)
+            setFeatures(style, truckRestrictions, congestion)
+          }
+        }
+        style.addEventListener('change', changeListener)
       } else {
-        map.removeLayer(defaultLayers.vector.normal.trafficincidents)
+        setFeatures(style, truckRestrictions, congestion)
       }
+    })
+  }, [defaultLayers, map, useVectorTiles, useSatellite, truckRestrictions, congestion])
+
+  useEffect(() => {
+    if (!map || !defaultLayers) {
+      return
     }
-  }, [incidentsLayer, map, defaultLayers, useVectorTiles])
+
+    if (!useVectorTiles) {
+      map.removeLayer(defaultLayers.hybrid.logistics.vector)
+      return
+    }
+
+    if (useSatellite) {
+      map.setBaseLayer(defaultLayers.hybrid.logistics.raster)
+      map.addLayer(defaultLayers.hybrid.logistics.vector)
+    } else {
+      map.setBaseLayer(defaultLayers.vector.normal.logistics)
+      map.removeLayer(defaultLayers.hybrid.logistics.vector)
+    }
+  }, [defaultLayers, map, useVectorTiles, useSatellite])
 
   useEffect(() => {
-    if (map && defaultLayers && useVectorTiles) {
-      if (trafficLayer) {
-        map.addLayer(defaultLayers.vector.normal.traffic)
-      } else {
-        map.removeLayer(defaultLayers.vector.normal.traffic)
-      }
+    if (!map || !defaultLayers) {
+      return
+    }
+
+    if (trafficLayer && useVectorTiles) {
+      map.addLayer(defaultLayers.vector.traffic.logistics)
+    } else {
+      map.removeLayer(defaultLayers.vector.traffic.logistics)
```

Comment:
nit: I guess we can combine this condition under the `useEffect` at line: 73

Reply:
I don't think we can do that because of this scenario:
- Initially: `trafficLayer` and `useSatellite` are both `false`
- User enables traffic:
  - `map.setBaseLayer(defaultLayers.vector.normal.logistics)` is also called (shouldn't be a huge problem)
  - `map.removeLayer(defaultLayers.hybrid.logistics.vector)` is also called (shouldn't be a huge problem)
  - `map.addLayer(defaultLayers.vector.traffic.logistics)` is called
- User also enables `useSatellite` (now both `trafficLayer` and `useSatellite` are enabled, hook is triggered again):
  - `map.setBaseLayer(defaultLayers.hybrid.logistics.raster)` is called
  - `map.addLayer(defaultLayers.hybrid.logistics.vector)` is called
  - `map.addLayer(defaultLayers.vector.traffic.logistics)` is called again, adding another traffic layer potentially causing issues